CRM/Admin/Form/Setting/Smtp.php

```diff
@@ -79,6 +80,14 @@ public function postProcess() {
 
     $formValues = $this->controller->exportValues($this->_name);
 
+    if (!empty($formValues['allow_mail_from_logged_in_contact'])) {
+      Civi::settings()->set('allow_mail_from_logged_in_contact', TRUE);
+    }
+    else {
+      Civi::settings()->set('allow_mail_from_logged_in_contact', FALSE);
+    }
+    unset($formValues['allow_mail_from_logged_in_contact']);
```

Comment:
This can be further optimised into 
```patch
diff --git a/CRM/Admin/Form/Setting/Smtp.php b/CRM/Admin/Form/Setting/Smtp.php
index e3219e8..d05ff48 100644
--- a/CRM/Admin/Form/Setting/Smtp.php
+++ b/CRM/Admin/Form/Setting/Smtp.php
@@ -80,12 +80,7 @@ class CRM_Admin_Form_Setting_Smtp extends CRM_Admin_Form_Setting {
 
     $formValues = $this->controller->exportValues($this->_name);
 
-    if (!empty($formValues['allow_mail_from_logged_in_contact'])) {
-      Civi::settings()->set('allow_mail_from_logged_in_contact', TRUE);
-    }
-    else {
-      Civi::settings()->set('allow_mail_from_logged_in_contact', FALSE);
-    }
+    Civi::settings()->set('allow_mail_from_logged_in_contact', (!empty($formValues['allow_mail_from_logged_in_contact'])));
```